pytest/conftest.py

```diff
@@ -84,7 +78,45 @@ def resource_with_metadata():
         owner=user,
         title=title,
         metadata=metadata,
-        files=(open('assets/cea.tif', 'rb'),)
+        files=(open('pytest.ini', 'rb'),)
     )
-    yield res_uuid
-    _res.delete()
+    return _res
+
+
+@pytest.mark.django_db
+@pytest.fixture(scope="function")
+def sample_user():
+    hydroshare_author_group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+    user = hydroshare.create_account(
+                '{}@noreply.org'.format(str(uuid.uuid4())),
+                username='{}'.format(str(uuid.uuid4())),
+                first_name='First',
+                last_name='Last',
+                superuser=False,
+                groups=[]
+            )
+    yield user
+    user.delete()
+
+
+@pytest.mark.django_db
+@pytest.fixture(scope="function")
+def public_resource_with_metadata():
+    resource = base_sample_resource()
+    resource.raccess.public = True
+    resource.keywords_string = str(uuid.uuid4())
+    resource.raccess.save()  # saves flag, doesn't necessarily re-index
+    resource.save()  # invokes re-indexing.
```

Comment:
ok, so resource save() invokes reindexin, but raccess save() does not.  That's good to know.

Reply:
I doubt it makes any difference, but it may be worth settings the resource discoverable through the set_discoverable method that hydroshare uses.  https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L1793

you would just have to make sure the resource has a title, keywords, abstract and a file.  It looks like you have everything except for the abstract, which you could add in the metadata dictionary where you set coverage, etc.

Reply:
I was told the save() invokes the reindex. The title, keyword, abstract are a frontend restriction.

Reply:
Like I said, I expect this works.  I'm just pointing out that a test that makes a resource public in the same way that the hydroshare app does, then the test will cover more edge cases.  If we're setting a resource public through a backdoor, there may be code that web app runs that the test does not, making potential bugs more obscure.